### ISSUE TITLE
Refactor max_supersaturation to use single solve + Hermite cubic evaluation

### ISSUE THREAD

**Issue by darothen** [darothen/pyrcel#79]
## Background

`max_supersaturation` currently triggers a *second* ODE solve via `_solve_dense`
(`SaveAt(dense=True)`), which uses a separate JIT-compiled kernel from the primary
`_solve` / `integrate_parcel` path.  This means two JIT kernels for what is
conceptually one simulation.

## Opportunity

The post-hoc `_peak_from_trajectory` added in #XX already demonstrates the key
idea: given saved trajectory data `(ts, ys)` from a **single** solve
(`SaveAt(ts=ts)`), we can recover S_max to O(Δt⁴) accuracy by:

1. Finding the coarse argmax index `si` on the saved grid.
2. Evaluating the ODE RHS at the three bracket points `(si-1, si, si+1)` — three
   eager calls.
3. Constructing `diffrax.ThirdOrderHermitePolynomialInterpolation` over each of
   the two sub-intervals `[si-1, si]` and `[si, si+1]` (passing `h * f` as `k0, k1`
   since the class expects `dp/du`, not `dy/dt`).
4. Finding the analytic maximum by solving the resulting quadratic `dp/du = 0`.

All of this runs inside JAX JIT with full differentiability.  The envelope theorem
(`stop_gradient` on `t_peak`) still applies for `d(S_max)/dV`.

## Proposed refactor

```
max_supersaturation(y0, args, ts)
```

1. Replace `_solve_dense` call with `_solve(y0, args, ts)` (same kernel as
   `integrate_parcel_chunked`).
2. After the solve, run a JAX-compatible version of the Hermite bracketing logic
   inside the JIT — using `jax.vmap` over the fine grid or analytic quadratic.
3. Return `stop_gradient(t_peak)` and `S(t_peak)` as before.

## Benefits

- **One JIT kernel** instead of two — `max_supersaturation` shares the compiled
  binary with `integrate_parcel`.
- **No `_solve_dense`** — eliminates the dense-output storage overhead.
- **Simpler code** — leverages `dfx.ThirdOrderHermitePolynomialInterpolation`
  (same class now used in `_peak_from_trajectory`) for evaluation.
- **Gradient accuracy unchanged** — envelope theorem still exact.

## Caveats / open questions

- The in-JIT Hermite bracketing needs to use `jax.lax` control flow (no Python
  `if/for`) or a fixed-size `jnp.argmax` + `jnp.where` idiom.
- The analytic quadratic `disc = B*B - 4*A*C` must use `jnp.where` to avoid
  NaN gradients when `disc < 0`.
- Need to verify gradient accuracy vs. the current `_solve_dense` approach.

## Related

- `_peak_from_trajectory` (introduced to fix `live=True, terminate=False` NaN bug)
  is the eager-Python prototype for this refactor.
- `diffrax.ThirdOrderHermitePolynomialInterpolation` is the correct class for
  cubic Hermite evaluation; it expects `k = h * dy/dt` as slope inputs.
